sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -84,6 +84,26 @@ int pollSocketUntilEventOrTimeout(
   return WSAPoll(&poller, 1, timeout);
 #endif
 }
+
+#ifdef WINDOWS
+// Windows needs this after every write to socket or peformance would be reduced to 1/4 for
+// uploading operation.
+// https://github.com/Azure/azure-sdk-for-cpp/issues/644
+void WinSocketSetBuffSize(curl_socket_t socket)
+{
+  ULONG ideal;
+  DWORD ideallen;
+  // WSAloctl would get the ideal size for the socket buffer.
+  if (WSAIoctl(socket, SIO_IDEAL_SEND_BACKLOG_QUERY, 0, 0, &ideal, sizeof(ideal), &ideallen, 0, 0)
```

Comment:
Capture the return code and log it.  We need better tracing throughout these code paths

Reply:
could be not such a good idea.
This method is called every time we write something to the wire. Usually, on WIndows, each write is 16Kb. We would add 65,536 log entry for uploading 1Mb, for instance

Reply:
This only needs to happen once per socket.  Why do we reset the socket buffer on each read/write?

Reply:
that's how libcurl does it XD.
This function is called on windows right after sending bits to server